astropy/io/votable/converters.py

```diff
@@ -674,19 +674,25 @@ def __init__(self, field, config=None, pos=None):
 
         precision = field.precision
         width = field.width
-        format_parts = ['%']
+
+        if precision is None:
+            format_parts = ['{0']
```

Comment:
I actually don't know if width can be a number in a real use case if precision is `None`, but the logic here allows it. If `%6r` (the old code) does not work anyway, then we don't have to worry about it.